pkg/manager/cert/cert_info.go

```diff
@@ -124,7 +126,12 @@ func (ci *certInfo) customizeTLSConfig(tlsConfig *tls.Config) *tls.Config {
 		}
 	} else {
 		tlsConfig.GetClientCertificate = func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			return ci.getCertificate(), nil
+			cert := ci.getCertificate()
+			if cert == nil {
```

Comment:
When is `cert == nil`? If the tlsConfig is nil, it won't come here. If the tlsConfig is not nil, can `cert` be nil?

Reply:
A tls config with only CA, or even without CA `insecureSkipVerify`.